apps/src/templates/teacherDashboard/googleClassroomRedux.js

```diff
@@ -0,0 +1,50 @@
+/* global gapi */
+
+const DISCOVERY = ['https://www.googleapis.com/discovery/v1/apis/classroom/v1/rest'];
+const CLIENT_ID = '254945981659-modh6ba8erd7iue9jr054o71dfa204o2.apps.googleusercontent.com';
+const SCOPE = [
+  'https://www.googleapis.com/auth/classroom.courses.readonly',
+  'https://www.googleapis.com/auth/classroom.rosters.readonly'
+].join(' ');
+
+const SET_CLASSROOM_LIST = 'teacherDashboard/SET_CLASSROOM_LIST';
+
+function getClassroomList(dispatch) {
+  gapi.client.classroom.courses.list({teacherId: 'me'}).then(response =>
+    dispatch(setClassroomList(response.result.courses))
+  );
+}
+
+export const loadClassroomList = () => {
+  return dispatch => {
+    gapi.load('client:auth2', () => {
+      gapi.client.init({
+        discoveryDocs: DISCOVERY,
+        clientId: CLIENT_ID,
+        scope: SCOPE,
+      }).then(() => {
+        if (gapi.auth2.getAuthInstance().isSignedIn.get()) {
+          getClassroomList(dispatch);
+        } else {
+          gapi.auth2.getAuthInstance().signIn().then(() => getClassroomList(dispatch));
+        }
+      });
+    });
+  };
+};
+export const setClassroomList = classrooms => ({ type: SET_CLASSROOM_LIST, classrooms });
+
+const initialState = {
+  classrooms: null,
+};
+
+export default function googleClassroom(state = initialState, action) {
+  if (action.type === SET_CLASSROOM_LIST) {
+    return {
+      ...state,
+      classrooms: action.classrooms,
+    };
+  }
+
+  return state;
+}
```

Comment:
I think I was more so questioning whether we actually need redux here is setting this state once. As I think some more, that still might make sense to do, so that we get the benefits of some of the async state handling (and can connect our components to update as soon as we have classrooms).

That said, classrooms is going to be the only thing in this reducer longterm, instead of having an object state, we could just return our set of classrooms, i.e.
```
if (action.type === SET_CLASSROOM_LIST) {
  return action.classRooms;
}
return state;
```
and then in our connect use `state.googleClassroom` instead of `state.googleClassroom.classroom`


If we're planning to have additional state/actions as part of this reducer in the future, then this is all moot and how you have it is great.